yktrace/src/swt.rs

```diff
@@ -137,19 +144,28 @@ unsafe extern "C" fn __yk_swt_rec_loc(symbol_name: *const i8, bb_idx: u32) {
     __yk_swt_rec_loc_impl(symbol_name, bb_idx);
 }
 
+// Cranelift produces suboptimal code for the fast path. When running on x86_64 and using ELF TLS
```

Comment:
This implementation *is* the fast path, isn't it?

In hindsight, I think something like: "An optimised trace recorder which avoids the cost of `__tls_get_addr`."

Reply:
This is not about avoiding the cost of `__tls_get_addr`. The linker already does that when possible. This is about avoiding the cost of unnecessary spilling of registers when tracing is disabled.

Reply:
So would something like this be accurate?

> Cranelift produces suboptimal code for the Rust version of `__yk_swt_rec_loc` (redundant spills). When running...